src/sql/workbench/browser/designer/designer.ts

```diff
@@ -941,6 +996,65 @@ export class Designer extends Disposable implements IThemable {
 		return component;
 	}
 
+	private addTableTaskbar(container: HTMLElement, tableProperties: DesignerTableProperties): Taskbar | undefined {
+		if (tableProperties.canAddRows || tableProperties.canMoveRows) {
+			const taskbarContainer = container.appendChild(DOM.$('.full-row')).appendChild(DOM.$('.add-row-button-container'));
+			const taskbar = new Taskbar(taskbarContainer);
+			let actions = [];
+			if (tableProperties.canAddRows) {
+				let addRowAction = this._instantiationService.createInstance(AddRowAction, this, tableProperties);
+				actions.push(addRowAction);
+			}
+			if (tableProperties.canMoveRows) {
+				let moveUpAction = this._instantiationService.createInstance(MoveRowUpAction, this);
+				let moveDownAction = this._instantiationService.createInstance(MoveRowDownAction, this);
+				actions.push(moveUpAction);
+				actions.push(moveDownAction);
+			}
+			let taskbarContent: ITaskbarContent[] = actions.map((a) => { return { action: a }; });
+			taskbar.setContent(taskbarContent);
+			this._actionsMap.set(taskbar, actions);
+			return taskbar;
+		}
+		return undefined;
+	}
+
+	private openContextMenu(table: Table<Slick.SlickData>, event: ITableMouseEvent, propertyPath: DesignerPropertyPath, view: DesignerUIArea): void {
+		const rowIndex = event.cell.row;
+		let tableActionContext: DesignerTableActionContext = {
+			table: table,
+			path: propertyPath,
+			source: view,
+			selectedRow: rowIndex
+		};
+		let data = table.grid.getData() as Slick.DataProvider<Slick.SlickData>;
+		if (!data || rowIndex >= data.getLength()) {
+			return undefined;
+		}
+		let actions = this.getTableActions();
+		actions.forEach(a => {
+			if (a instanceof DesignerTableAction) {
+				a.table = table;
+				a.updateState(rowIndex);
+			}
+		});
+		this._contextMenuService.showContextMenu({
+			getAnchor: () => event.anchor,
+			getActions: () => actions,
+			getActionsContext: () => (tableActionContext)
+		});
+	}
+
+	private getTableActions(): IAction[] {
+		let actions: IAction[];
+		let insertRowBefore = this._instantiationService.createInstance(InsertBeforeSelectedRowAction, this);
```

Comment:
we have to check canInsertRows and canMoveRows and decide what to add to the context menu. and please also make sure only the enabled actions are added.